### PR TITLE
CLDC-3788 Export sales logs

### PR DESCRIPTION
**Pull request by kosiakkatrina**

### CODE REVIEW

File: app/services/exports/sales_log_export_service.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        SalesLog.exportable.where("(updated_at >= :from AND updated_at <= :to) OR (values_updated_at IS NOT NULL AND values_updated_at >= :from AND values_updated_at <= :to)", params).filter_by_year(year)
+      else
+        params = { to: @start_time }
+        SalesLog.exportable.where("updated_at <= :to", params).filter_by_year(year)
+      end
+    end
+
+    def apply_cds_transformation(sales_log, _export_mode)
+      sales_log.attributes_before_type_cast
+      # attribute_hash["formid"] = attribute_hash["old_form_id"] || (attribute_hash["id"] + LOG_ID_OFFSET)
```
Comment (RachaelBooth) [inline app/services/exports/sales_log_export_service.rb]: Can this go?

File: app/services/exports/export_service.rb [hunk 2]
```diff
@@ -20,16 +21,19 @@ def export_xml(full_update: false, collection: nil, year: nil)
...
         end
       else
         users_archives_for_manifest = get_user_archives(start_time, full_update)
         organisations_archives_for_manifest = get_organisation_archives(start_time, full_update)
         lettings_archives_for_manifest = get_lettings_archives(start_time, full_update, year)
+        sales_archives_for_manifest = get_sales_archives(start_time, full_update, year) if Time.zone.now >= Time.zone.local(2025, 4, 1)
       end
 
-      write_master_manifest(daily_run_number, lettings_archives_for_manifest.merge(users_archives_for_manifest).merge(organisations_archives_for_manifest))
+      write_master_manifest(daily_run_number, lettings_archives_for_manifest.merge(sales_archives_for_manifest || {}).merge(users_archives_for_manifest).merge(organisations_archives_for_manifest))
```
Comment (RachaelBooth) [inline app/services/exports/export_service.rb]: Will `sales_archives_for_manifest` not be `{}` by default anyway?

File: spec/services/exports/sales_log_export_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+  let(:expected_zip_filename) { "core_sales_2025_2026_apr_mar_f0001_inc0001.zip" }
+  let(:expected_data_filename) { "core_sales_2025_2026_apr_mar_f0001_inc0001_pt001.xml" }
+  let(:expected_manifest_filename) { "manifest.xml" }
+  let(:start_time) { Time.zone.local(2026, 3, 1) }
+  let(:organisation) { create(:organisation, name: "MHCLG", housing_registration_no: 1234) }
+  let(:user) { FactoryBot.create(:user, email: "test1@example.com", organisation:) }
+
+  def replace_entity_ids(sales_log, export_template)
+    # export_template.sub!(/\{id\}/, (sales_log["id"] + Exports::SalesLogExportService::LOG_ID_OFFSET).to_s)
```
Comment (RachaelBooth) [inline spec/services/exports/sales_log_export_service_spec.rb]: Does this need to be here?

**Review by RachaelBooth — APPROVED** [review]

File: app/services/exports/sales_log_export_service.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/services/exports/sales_log_export_service.rb]: Yep, I think I kept it there as a reminder cause we don't have exact field descriptions (ids included)

File: app/services/exports/export_service.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/services/exports/export_service.rb]: Ah, yep, good point

**Pull request closed without merging**